source/dofs/dof_renumbering.cc

```diff
@@ -730,13 +736,13 @@ namespace DoFRenumbering
     const types::global_dof_index result =
       compute_component_wise<dim, spacedim>(
         renumbering, start, end, component_order_arg, true);
+    (void)result;
 
-    if (result == 0)
-      return;
-
-    Assert(result == dof_handler.n_dofs(level), ExcInternalError());
+    Assert(result == 0 || result == dof_handler.n_dofs(level),
+           ExcInternalError());
 
-    if (renumbering.size() != 0)
+    if (Utilities::MPI::max(renumbering.size(),
+                            dof_handler.get_communicator()) > 0)
```

Comment:
This and the similar code below was a bug, right? It would fail if a process has no locally owned DoFs.

Reply:
Yes.